service/paging.go

```diff
@@ -0,0 +1,64 @@
+package service
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+)
+
+// PagingParams represents paging and sorting parameter values
+type PagingParams struct {
+	PageSize   *int32
+	PageNumber *int32
+	SortField  string
+	SortAsc    bool
+}
+
+// ParsePagingParams retrieves paging params from the request, allows for whitelisting sort field values
+func ParsePagingParams(r *http.Request, sortFieldsWhitelist []string) (PagingParams, error) {
+	paging := PagingParams{}
+
+	// page number
+	pageNumber, err := Int32PointerFromQueryParam(r, "page_number")
+	if err != nil {
+		return paging, err
+	}
+	paging.PageNumber = pageNumber
+
+	// page size
+	pageSize, err := Int32PointerFromQueryParam(r, "page_size")
+	if err != nil {
+		return paging, err
+	}
+	paging.PageSize = pageSize
+
+	// sort asc
+	sortAscStr := r.URL.Query().Get("sort_asc")
+	if len(sortAscStr) > 0 {
```

Comment:
What if sort_asc is not included? There isn't a test. len(nil) would throw an exception.  Do you want that or to not fail?

Reply:
`Get` would return an empty string